src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -80,9 +128,30 @@ static void create_data(ProgramBuilder& p, const ov::Shape& const_shape, const s
         constTensor.feature[0] = 1;
     }
 
+    cldnn::padding padding_data = cldnn::padding();
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    if (p.get_config().get_use_onednn()
+            && (op->get_element_type() == ov::element::i4 || op->get_element_type() == ov::element::u4)
+            && !op->is_dynamic()
+            && const_shape.size() == 2) {
```

Comment:
we dont' need to limit the case as shape_size == 2. we may have batch, feature, y => 3D.

Reply:
Fixed code to support 3D/4D  and made ocl kernel instead of CPU computation on constant creation. please review it again. thanks.